### DISCUSSION TITLE
Implementation of PML boundaries in RZ

### DISCUSSION THREAD

**Discussion by OscarJakobssonLPA** [BLAST-WarpX/warpx#2076] [Simulation Inputs] [answered]
Dear developers, 

In issue #904 the implementation of PML boundaries in the RZ geometry is discussed. There are several scenarios that our group is interested in for which this would be very useful, for either the FDTD or PSATD schemes. For instance, we have found that electron acceleration in leakage plasma waveguides during long multi-GeV stages, for which 3D simulations are costly, is sensitive to reflections of the walls of the simulation box.  

Is this likely to be a feature of WarpX in the near future? 

Thanks in advance, 
Oscar

**Comment by jlvay** [comment, marked as answer]
For now, we have only the Silver-Mueller condition that is implemented in RZ for FDTD. Implementing PMLs for FDTD and PSATD in RZ is certainly doable and on our  long-term to-do list but it is hard to predict at this stage when it will be implemented. We can help anyone who would like to do the implementation before we get to it.

**Reply by OscarJakobssonLPA** [reply]
Thank you @jlvay for you reply.
I will have a look at the PML implementation in FBPIC and see what I can do. 
If I can find the time I would be happy to give it a go.
